### PR TITLE
Update OV / NNCF/ ModelAPI / Datumaro

### PR DESCRIPTION
**Pull request by sovrasov**
### Summary
This PR updates OV-related dependencies of OTX.
Known issues:
- OMZ plugin may fail in some cases, because version was set to opset1 by default for some of operations. This could be updated later, if we face models that show opset - version mismatch.
- ModelAPI integration requires a deeper refactoring
- MoViNet inference in OV doesn't work
- Tile classifier is not handled by PTQ (as before)

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [x] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by sungmanc** [comment]
Small comment, if the PR is still in progress, please use DRAFT PR. we could use CI tests even in under the DRFAT status.

**Comment by yunchu** [comment]
@sovrasov note that you can skip the automatic trigger of the actions for the test PR by adding string [skip ci] to your commit message.
https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

File: otx/algorithms/action/adapters/openvino/task.py [hunk 1]
```diff
@@ -18,25 +18,26 @@
...
 
+import nncf
 import numpy as np
-from addict import Dict as ADDict
-from compression.api import DataLoader
-from compression.engines.ie_engine import IEEngine
-from compression.graph import load_model, save_model
-from compression.graph.model_utils import compress_model_weights, get_nodes_by_type
-from compression.pipeline.initializer import create_pipeline
+import openvino.runtime as ov
```
Comment (sungmanc) [inline otx/algorithms/action/adapters/openvino/task.py]: minor thing: how about `ovr` instead of `ov`?

File: tests/e2e/cli/action/test_action_classification.py [hunk 2]
```diff
@@ -96,6 +96,8 @@ def test_otx_export(self, template, tmp_dir_path):
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
     def test_otx_eval_openvino(self, template, tmp_dir_path):
         tmp_dir_path = tmp_dir_path / "action_cls"
+        if template.model_template_id == "Custom_Action_Classification_MoViNet":
+            pytest.xfail("CVS-112824: MoViNet inference fails in OV 2023.0")
```
Comment (sungmanc) [inline tests/e2e/cli/action/test_action_classification.py]: Could you made a GH issue and please use the GH number instead of CVS number since OTX is open-source

File: tests/integration/cli/action/test_action_classification.py [hunk 3]
```diff
@@ -108,6 +108,8 @@ def test_otx_export_onnx(self, template, tmp_dir_path):
     @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
     def test_otx_eval_openvino(self, template, tmp_dir_path):
+        if template.model_template_id == "Custom_Action_Classification_MoViNet":
+            pytest.xfail("CVS-112824: MoViNet inference fails in OV 2023.0")
```
Comment (sungmanc) [inline tests/integration/cli/action/test_action_classification.py]: Same here

File: tests/unit/algorithms/detection/adapters/openvino/model_wrappers/test_detection_openvino_models.py [hunk 4]
```diff
@@ -74,6 +74,7 @@ def __init__(self, *args):
         self.confidence_threshold = 0.375
         self.resize_type = "standard"
         self.output_parser = MockBatchBoxesLabelsParser()
+        self.labels = ["a"]
```
Comment (sungmanc) [inline tests/unit/algorithms/detection/adapters/openvino/model_wrappers/test_detection_openvino_models.py]: Just wondering, is it necessary?

**Review by sungmanc — COMMENTED** [review]
Thanks for the nice progress, please check my comments

File: tests/unit/algorithms/detection/adapters/openvino/model_wrappers/test_detection_openvino_models.py (hunk 4 above)
Comment (sovrasov) [reply tests/unit/algorithms/detection/adapters/openvino/model_wrappers/test_detection_openvino_models.py]: New model API requires this attribute to be added. An empty list is also ok, so I'll change it.

File: otx/algorithms/action/adapters/openvino/task.py (hunk 1 above)
Comment (sovrasov) [reply otx/algorithms/action/adapters/openvino/task.py]: We have ov everywhere in omz plugin. That change requires refactoring of ~ 50 files.

File: otx/algorithms/action/adapters/openvino/task.py [hunk 5]
```diff
@@ -147,20 +141,26 @@ def forward(self, image: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
         return self.model.infer_sync(image)
 
 
-class DataLoaderWrapper(DataLoader):
+class DataLoaderWrapper:
     """DataLoader implementation for ActionOpenVINOTask."""
 
-    def __init__(self, dataloader: DataLoader, inferencer: BaseInferencer):
-        super().__init__(config=None)
+    def __init__(self, dataloader: Any, inferencer: BaseInferencer, shuffle: bool = True):
```
Comment (jaegukhyun) [inline otx/algorithms/action/adapters/openvino/task.py]: Normally openvino tasks are used to inference. Therefore, I think default value for shuffle should be False.
What do you think?

File: otx/algorithms/classification/adapters/openvino/model_wrappers/openvino_models.py [hunk 6]
```diff
@@ -86,18 +84,18 @@ def _get_outputs(self):
 
     def postprocess(self, outputs: Dict[str, np.ndarray], meta: Dict[str, Any]):  # pylint: disable=unused-argument
         """Post-process."""
-        logits = outputs[self.out_layer_name].squeeze()
+        logits = outputs[self.out_layer_names[-1]].squeeze()
```
Comment (jaegukhyun) [inline otx/algorithms/classification/adapters/openvino/model_wrappers/openvino_models.py]: Did you check backward compatibility for classification ov inference? 
For example 
1. Export in otx1.3
2. Inference exported openvino.xml in this branch

File: tests/unit/algorithms/detection/adapters/openvino/test_task.py [hunk 7]
```diff
@@ -234,22 +233,22 @@ def test_deploy(self, otx_model):
...
         output_model = copy.deepcopy(otx_model)
         self.ov_task.model.set_data("openvino.bin", b"foo")
         self.ov_task.model.set_data("openvino.xml", b"bar")
-        mocker.patch("otx.algorithms.detection.adapters.openvino.task.load_model", autospec=True)
-        mocker.patch("otx.algorithms.detection.adapters.openvino.task.create_pipeline", autospec=True)
-        mocker.patch("otx.algorithms.detection.adapters.openvino.task.save_model", new=patch_save_model)
-        spy_compress = mocker.spy(otx.algorithms.detection.adapters.openvino.task, "compress_model_weights")
+        mocker.patch("otx.algorithms.segmentation.adapters.openvino.task.ov.Core.read_model", autospec=True)
+        mocker.patch("otx.algorithms.segmentation.adapters.openvino.task.ov.serialize", new=patch_save_model)
+        fake_quantize = mocker.patch("otx.algorithms.segmentation.adapters.openvino.task.nncf.quantize", autospec=True)
```
Comment (jaegukhyun) [inline tests/unit/algorithms/detection/adapters/openvino/test_task.py]: This is test for detection task, but patching segmentation's adapters. Is it intended?

File: requirements/base.txt [hunk 8]
```diff
@@ -4,7 +4,7 @@ natsort>=6.0.0
 prettytable
 protobuf>=3.20.0
 pyyaml
-datumaro==1.2.0rc3
+datumaro@ git+https://github.com/openvinotoolkit/datumaro@e9e5307efdbd51598cbfc9fe3fd7c282870b98ef#egg=datumaro
```
Comment (jaegukhyun) [inline requirements/base.txt]: Should we enforce latest branch for datumaro?

File: otx/core/ov/ops/op.py [hunk 9]
```diff
@@ -32,7 +32,7 @@ class Operation(torch.nn.Module, Generic[_T]):  # pylint: disable=abstract-metho
     """Operation class."""
 
     TYPE = ""
-    VERSION = -1
+    VERSION = ""
```
Comment (jaegukhyun) [inline otx/core/ov/ops/op.py]: Just curious, what is the meaning of VERSION?

File: otx/algorithms/common/utils/utils.py [hunk 10]
```diff
@@ -126,3 +129,27 @@ def get_default_async_reqs_num() -> int:
         return reqs_num
     else:
         return 1
+
+
+def read_py_config(filename: str) -> adict:
+    """Reads py condig to a dict."""
```
Comment (jaegukhyun) [inline otx/algorithms/common/utils/utils.py]:
```suggestion
    """Reads py config to a dict."""
```

File: otx/algorithms/detection/adapters/mmdet/task.py [hunk 11]
```diff
@@ -396,8 +396,6 @@ def hook(module, inp, outp):
             saliency_hook: Union[nullcontext, BaseRecordingForwardHook] = nullcontext()
         else:
             raw_model = feature_model
-            if raw_model.__class__.__name__ == "NNCFNetwork":
-                raw_model = raw_model.get_nncf_wrapped_model()
```
Comment (jaegukhyun) [inline otx/algorithms/detection/adapters/mmdet/task.py]: Could you explain why we can remove this line?

**Review by jaegukhyun — COMMENTED** [review]
First of all, thanks for your huge effort.
Since it is a quite large PR, I just skim whole PR. I left some comments, change suggestions, and questions.

And here are some general questions.
1. Did you check your own regression test?
For example, when you train model on otx1.3 and export model both on otx1.3 and your branch,
does the exported model show same or better accuracy and same or better speed?
2. Did you check your own backward compatibility test?
For example, when you train and export model on otx1.3 and run inference the exported model in your branch, is there any error or issue?

Maybe this kind of tests for all tasks would be tough for you. If you need help for other colleagues, feel free to ask.

File: otx/algorithms/action/adapters/openvino/task.py (hunk 5 above)
Comment (sovrasov) [reply otx/algorithms/action/adapters/openvino/task.py]: This dataloader is used for PTQ optimize only, when we have to manually apply preprocessing.
By design it shouldn't be applied outside of PTQ optimization, so I set this field to true for compliance with old POT config.

File: otx/algorithms/classification/adapters/openvino/model_wrappers/openvino_models.py (hunk 6 above)
Comment (sovrasov) [reply otx/algorithms/classification/adapters/openvino/model_wrappers/openvino_models.py]: I checked that for classification, but it was rather for debugging purposed.

This PR doesn't bring any changes to openvino.xml (excepting ones that new MO could introduce). All changes in OV wrappers are related to upgrade of ModelAPI version, so old IRs are handled as well (since input/output specs like names and shapes were not changed).

File: otx/core/ov/ops/op.py (hunk 9 above)
Comment (sovrasov) [reply otx/core/ov/ops/op.py]: It represents opset in terms of openvino: opset1, opset2, etc. Some of ops still have integer opset values. You may manually inspect openvino.xml you'll find it.
Unfortunately, now we have no resources to update all the ops with the correct opset versions, but I did that for the ones, that are used in tests.

File: requirements/base.txt (hunk 8 above)
Comment (sovrasov) [reply requirements/base.txt]: I don't see a reason for that, unless we need to use the latest features. Setting develop here would lead to rolling upgrade of datumaro and may cause unpredictable effect in tests.
Prior to OTX release, this requirement should be changed to a package version (which doesn't exist yet).

File: tests/unit/algorithms/detection/adapters/openvino/test_task.py (hunk 7 above)
Comment (sovrasov) [reply tests/unit/algorithms/detection/adapters/openvino/test_task.py]: No it's a copy-paste error

File: otx/algorithms/detection/adapters/mmdet/task.py (hunk 11 above)
Comment (sovrasov) [reply otx/algorithms/detection/adapters/mmdet/task.py]: That happened because of changes in NNCF. They improved the process of wrapping, so get_nncf_wrapped_model() method is no longer required:
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/nncf_network.py#LL980C11-L980C11

File: tests/integration/cli/action/test_action_classification.py (hunk 3 above)
Comment (sovrasov) [reply tests/integration/cli/action/test_action_classification.py]: Updated, we already have an issue on github

File: tests/e2e/cli/action/test_action_classification.py (hunk 2 above)
Comment (sovrasov) [reply tests/e2e/cli/action/test_action_classification.py]: Updated, we already have an issue on github

File: otx/algorithms/common/utils/utils.py (hunk 10 above)
Comment (sovrasov) [reply otx/algorithms/common/utils/utils.py]: Fixed

**Comment by sovrasov** [comment]
> First of all, thanks for your huge effort. Since it is a quite large PR, I just skim whole PR. I left some comments, change suggestions, and questions.
> 
> And here are some general questions.
> 
> 1. Did you check your own regression test?
>    For example, when you train model on otx1.3 and export model both on otx1.3 and your branch,
>    does the exported model show same or better accuracy and same or better speed?
> 2. Did you check your own backward compatibility test?
>    For example, when you train and export model on otx1.3 and run inference the exported model in your branch, is there any error or issue?
> 
> Maybe this kind of tests for all tasks would be tough for you. If you need help for other colleagues, feel free to ask.

1. I conducted some investigations of OV inference accuracy/performance. Overall performance of IR inference is carefully tested on OV CPU plugin's side, and I didn't see any noticeable degradation (I did a little comparison when I was working on async inference). The situation with accuracy is more complicated: there are some changes introduced by the new model API, that cause accuracy difference with the previous version, and I still didn't locate the source of it. I launched e2e tests, and accuracy change fits deltas for all models, excepting yolox. I'll conduct a standalone investigation on that model. Also, PTQ/NNCF upgrade would bring some changes to accuracy of compressed models. We'll see the exact numbers after launch of regression tests.
2. I use pretty old IRs (from OTX 1.1, 1.2) in some cases, and already described why backward compatibility of .xml files shouldn't be an issue, in my opinion.

File: otx/algorithms/anomaly/tasks/openvino.py [hunk 12]
```diff
@@ -276,16 +278,17 @@ def optimize(
             ValueError: When the optimization type is not POT, which is the only support type at the moment.
         """
         if optimization_type is not OptimizationType.POT:
-            raise ValueError("POT is the only supported optimization type for OpenVINO models")
+            raise ValueError("PTQ is the only supported optimization type for OpenVINO models")
```
Comment (ashwinvaidya17) [inline otx/algorithms/anomaly/tasks/openvino.py]: What is PTQ? And should we change `OptimizationType.POT` to `OptimizationType.PTQ`

**Comment by sovrasov** [comment]
@eugene123tw please have a look at IS tiling inference fix

File: otx/core/ov/ops/utils.py [hunk 13]
```diff
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: MIT
```
Comment (ashwinvaidya17) [inline otx/core/ov/ops/utils.py]: Not exactly related to this PR. But is `MIT` licence correct?

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for the efforts. These are a lot of files. I didn't go through all of them very thoroughly but the ones related to anomaly tasks look good to me. I just have one question regarding PTQ.

File: otx/algorithms/anomaly/tasks/openvino.py (hunk 12 above)
Comment (sovrasov) [reply otx/algorithms/anomaly/tasks/openvino.py]: PTQ is a tool to replace POT, which is deprecated. Changing OptimizationType requires corresponding changes on Geti's side. I decided to postpone any breaking changes for now to simplify the integration.

File: otx/core/ov/ops/utils.py (hunk 13 above)
Comment (sovrasov) [reply otx/core/ov/ops/utils.py]: I guess no, it was added occasionally because of a mistake in out PR template. I'll create an issue to eliminate all wrong licenses.

Comment (sovrasov) [reply otx/core/ov/ops/utils.py]: https://github.com/openvinotoolkit/training_extensions/issues/2259

File: otx/algorithms/anomaly/tasks/openvino.py (hunk 12 above)
Comment (ashwinvaidya17) [reply otx/algorithms/anomaly/tasks/openvino.py]: Thanks for the clarification

File: otx/algorithms/action/adapters/openvino/task.py (hunk 5 above)
Comment (jaegukhyun) [reply otx/algorithms/action/adapters/openvino/task.py]: Ah, my mistakes. Thanks for the clarification.

**Comment by jaegukhyun** [comment]
> > First of all, thanks for your huge effort. Since it is a quite large PR, I just skim whole PR. I left some comments, change suggestions, and questions.
> > And here are some general questions.
> > 
> > 1. Did you check your own regression test?
> >    For example, when you train model on otx1.3 and export model both on otx1.3 and your branch,
> >    does the exported model show same or better accuracy and same or better speed?
> > 2. Did you check your own backward compatibility test?
> >    For example, when you train and export model on otx1.3 and run inference the exported model in your branch, is there any error or issue?
> > 
> > Maybe this kind of tests for all tasks would be tough for you. If you need help for other colleagues, feel free to ask.
> 
> 1. I conducted some investigations of OV inference accuracy/performance. Overall performance of IR inference is carefully tested on OV CPU plugin's side, and I didn't see any noticeable degradation (I did a little comparison when I was working on async inference). The situation with accuracy is more complicated: there are some changes introduced by the new model API, that cause accuracy difference with the previous version, and I still didn't locate the source of it. I launched e2e tests, and accuracy change fits deltas for all models, excepting yolox. I'll conduct a standalone investigation on that model. Also, PTQ/NNCF upgrade would bring some changes to accuracy of compressed models. We'll see the exact numbers after launch of regression tests.
> 2. I use pretty old IRs (from OTX 1.1, 1.2) in some cases, and already described why backward compatibility of .xml files shouldn't be an issue, in my opinion.

Thanks for the info, I'll check some backward compatibility and performance regression on action and detection tasks.
And regarding to YOLOX, all performance delta value are adjusted on IK A100 server. Therefore you can see failures in your machine due to GPU difference. Therefore I recommend to check the performance degrade on A100 server or just check in your machine with large dataset(sufficient training)

**Comment by jaegukhyun** [comment]
Here's the checking results for detection.
BCCD dataset
  - Exported on OTX1.3
    ATSS
      - 1.3: 0.7983789260385005, 0.04972231277820861 secs
      - New: 0.8020304568527918, 0.03207170527700024 secs
    SSD
      - 1.3: 0.7805383022774326, 0.03563572355940576 secs
      - New: 0.7821576763485475, 0.021919174719692416 secs
    YOLOX 
      - 1.3: 0.7361835245046923, 0.008346731721151931 secs
      - New: 0.7369519832985385, 0.006730128417580595 secs
  - Exported with OV2023.0
    ATSS: 0.8020304568527918, 0.03207170527700024 secs
    SSD: 0.7821576763485475, 0.021919174719692416 secs
    YOLOX:0.7369519832985385, 0.006848513273224752 secs

In conclusion, after applying this PR, the exported xml is same as before. However, with same openvino xml, this PR makes more accurate and faster inference. It is very impressing.

One more thing, it seems that we can now export two stage detector well. Could you delete skip mark of test for openvino inference for using two-stage detector?
Here's the list
image

**Comment by sovrasov** [comment]
> Here's the checking results for detection. BCCD dataset
> 
> * Exported on OTX1.3
>   ATSS
>   
>   * 1.3: 0.7983789260385005, 0.04972231277820861 secs
>   * New: 0.8020304568527918, 0.03207170527700024 secs
>     SSD
>   * 1.3: 0.7805383022774326, 0.03563572355940576 secs
>   * New: 0.7821576763485475, 0.021919174719692416 secs
>     YOLOX
>   * 1.3: 0.7361835245046923, 0.008346731721151931 secs
>   * New: 0.7369519832985385, 0.006730128417580595 secs
> * Exported with OV2023.0
>   ATSS: 0.8020304568527918, 0.03207170527700024 secs
>   SSD: 0.7821576763485475, 0.021919174719692416 secs
>   YOLOX:0.7369519832985385, 0.006848513273224752 secs
> 
> In conclusion, after applying this PR, the exported xml is same as before. However, with same openvino xml, this PR makes more accurate and faster inference. It is very impressing.
> 
> One more thing, it seems that we can now export two stage detector well. Could you delete skip mark of test for openvino inference for using two-stage detector? Here's the list image

Thanks for double-checking! I removed skip marks for 2 stage detectors.

**Comment by jaegukhyun** [comment]
A link for e2e tests 
https://github.com/openvinotoolkit/training_extensions/actions/runs/5351939622

**Comment by sovrasov** [comment]
> A link for e2e tests https://github.com/openvinotoolkit/training_extensions/actions/runs/5351939622

All the re-enabled E2E openvino_eval tests were failed for 2 stage detectors, also YOLOX accuracy is below the target, as I said earlier. Additional investigation is required

**Comment by jaegukhyun** [comment]
> > A link for e2e tests https://github.com/openvinotoolkit/training_extensions/actions/runs/5351939622
> 
> All the re-enabled E2E openvino_eval tests were failed for 2 stage detectors, also YOLOX accuracy is below the target, as I said earlier. Additional investigation is required

It is weird, that all two stage methods show 0.0 accuracy. In local, they shows proper accuracy. Could you check tox.ini or requirements.txt?

**Comment by sovrasov** [comment]
> > > A link for e2e tests https://github.com/openvinotoolkit/training_extensions/actions/runs/5351939622
> > 
> > 
> > All the re-enabled E2E openvino_eval tests were failed for 2 stage detectors, also YOLOX accuracy is below the target, as I said earlier. Additional investigation is required
> 
> It is weird, that all two stage methods show 0.0 accuracy. In local, they shows proper accuracy. Could you check tox.ini or requirements.txt?

I faced the same problem in my local environment when launching e2e tests without tox. In non-e2e cases I see non-zero performance.

**Review by eugene123tw — DISMISSED** [review]
LGTM. Thanks Vlad!

**Review by sungmanc — DISMISSED** [review]
Thanks for the huge efforts !

**Review by jaegukhyun — CHANGES_REQUESTED** [review]
I think we should solve e2e test failure issue before merging this PR.

**Comment by sovrasov** [comment]
@jaegukhyun issues with IS accuracy we

> I think we should solve e2e test failure issue before merging this PR.

MaskRCNN wrapper was updated to comply with the new ModelAPI, but it didn't resolve the issue. Previously, the accuracy was 0 in all tests, but now some tests are passing (the same accuracy), in some others accuracy is less than target.

YOLOX test still fails. I managed to reproduce that failure on develop (I just trained the model for a different number of epochs), so I think it's rather some kind of instability in OV, that happens regardless of version.

**Comment by jaegukhyun** [comment]
https://github.com/openvinotoolkit/training_extensions/actions/runs/5396476722
Daily run for final commit [fedbab2](https://github.com/openvinotoolkit/training_extensions/pull/2090/commits/fedbab2c182e1e9fa1ca4723f2a701bfeeeb85b4)

File: otx/api/usecases/exportable_code/demo/requirements.txt [hunk 14]
```diff
@@ -1,4 +1,4 @@
-openvino==2022.3.0
-openmodelzoo-modelapi==2022.3.0
-otx @ git+https://github.com/openvinotoolkit/training_extensions/@a70d8f486f2dab289dffb6879606a2dfd827964d#egg=otx
+openvino==2023.0
```
Comment (eugene123tw) [inline otx/api/usecases/exportable_code/demo/requirements.txt]: @sovrasov Could you also check if there's a noticeable accuracy drop on instance segmentation? Recently someone reported this issue: https://github.com/openvinotoolkit/training_extensions/issues/2252#issue-1763838906. 

That being said, I still notice a large accuracy drop after installing 2023.0 with PyTorch 1.13.x. Could you verify if this is no longer an issue for PyTorch 1.13.x after the upgrade?

Comment (sovrasov) [reply otx/api/usecases/exportable_code/demo/requirements.txt]: @eugene123tw @jaegukhyun here are my results on coco_subset_cls_decr (496 imgs in val) from regression tests:
| Model  | pytorch 1.13 | OV2023 + new model API   |
|---|---|---|
| efficientnetb2b_maskrcnn  | 0.3488 | 0.3546  | 
| resnet50_maskrcnn  |  0.5212 | 0.5170  |

File: otx/algorithms/common/adapters/mmcv/nncf/utils.py [hunk 15]
```diff
@@ -117,6 +117,30 @@ def model_eval(
         return eval_res[metric_name]
 
 
+def nncf_state_dict_pre_hook(state_dict, prefix, *args, **kwargs):
+    """NNCF-specific state dict pre-hook."""
```
Comment (nikita-savelyevv) [inline otx/algorithms/common/adapters/mmcv/nncf/utils.py]: Could you please describe the reason this hook is needed? It may be helpful in the future.

File: otx/algorithms/common/adapters/mmcv/nncf/utils.py [hunk 16]
```diff
@@ -117,6 +117,30 @@ def model_eval(
...
+        if "_nncf" in key:
+            if key.startswith("backbone"):
+                key = key.replace("backbone.", "", 1)
+        state_dict[key] = val
+
+    return state_dict
+
+
+def nncf_state_dict_hook(module, state_dict, prefix, *args, **kwargs):
+    """NNCF-specific state dict post-hook."""
```
Comment (nikita-savelyevv) [inline otx/algorithms/common/adapters/mmcv/nncf/utils.py]: Same

**Review by nikita-savelyevv — COMMENTED** [review]
Great work! Some minor comments from my side.

**Review by nikita-savelyevv — DISMISSED** [review]

**Comment by jaegukhyun** [comment]
E2E test failure for action detection is not reproduced in public dataset. Therefore, we can assume that 0.0 exported model only happen in toy dataset. I created https://github.com/openvinotoolkit/training_extensions/issues/2279. Could you generate skip mark for action detection e2e?

**Comment by sovrasov** [comment]
> E2E test failure for action detection is not reproduced in public dataset. Therefore, we can assume that 0.0 exported model only happen in toy dataset. I created #2279. Could you generate skip mark for action detection e2e?

Action detection tests are skipped

File: tests/regression/instance_segmentation/test_instance_segmentation.py [hunk 17]
```diff
@@ -172,7 +172,7 @@ def test_otx_train_cls_incr_kpi_test(self, template):
 
     @e2e_pytest_component
     @pytest.mark.parametrize("template", templates, ids=templates_ids)
-    @pytest.mark.skip(reason="Issue#2059: Two stage detector shows 0.0 mAP when OpenVINO exported in torch 1.13")
+    @pytest.mark.skip(reason="Issue#2290: MaskRCNN shows degraded performance when inferencing in OpenVINO")
```
Comment (jaegukhyun) [inline tests/regression/instance_segmentation/test_instance_segmentation.py:175]: As far as I know, exporting model trained on large dataset doesn't show performance drop. Please correct me if I'm wrong

Comment (sovrasov) [reply tests/regression/instance_segmentation/test_instance_segmentation.py:175]: It's contradictive. Sometimes a minor drop happens, like with action detection

**Comment by jaegukhyun** [comment]
https://github.com/openvinotoolkit/training_extensions/actions/runs/5442410680
E2E Test from f8f3d67. Could you take a look? I have no idea for anomaly, but for detection, maybe data pipeline between normal detection and tiling would be different. Therefore [eeabbca](https://github.com/openvinotoolkit/training_extensions/pull/2090/commits/eeabbca84acafef156ec6072c1530ef3fed61ce3) could affect it.

**Comment by sovrasov** [comment]
> https://github.com/openvinotoolkit/training_extensions/actions/runs/5442410680 E2E Test from [f8f3d67](https://github.com/openvinotoolkit/training_extensions/commit/f8f3d67c5329164d6c78967fa8fd2f8bddde8b1e). Could you take a look? I have no idea for anomaly, but for detection, maybe data pipeline between normal detection and tiling would be different. Therefore [eeabbca](https://github.com/openvinotoolkit/training_extensions/pull/2090/commits/eeabbca84acafef156ec6072c1530ef3fed61ce3) could affect it.

In the latest commits, PTQ was switched to POT via config (NCT team members suggested to do that).
Also, the issue with YOLOX tiling preprocessing was fixed.

**Comment by sovrasov** [comment]
@jaegukhyun could you launch e2e once again? I was unable to reproduce the anomaly deploy accuracy issue on my side

**Comment by jaegukhyun** [comment]
https://github.com/openvinotoolkit/training_extensions/actions/runs/5460287541 I rerun e2e from [deea851](https://github.com/openvinotoolkit/training_extensions/pull/2090/commits/deea851c5123f0a2859e64a5d30e4c7011f6c4b9)
Firstly @GalyaZalesskaya should solve current E2E and @sovrasov you need to update FQ reference for anomaly and segmentation

**Comment by jaegukhyun** [comment]
Please rebase your branch, Galina's PR for e2e test failures is merged

**Review by jaegukhyun — DISMISSED** [review]
https://github.com/openvinotoolkit/training_extensions/actions/runs/5473307857
All tests are passed, I think we can merge this PR. Thanks for your effort

File: CHANGELOG.md [hunk 18]
```diff
@@ -30,7 +31,7 @@ All notable changes to this project will be documented in this file.
 
 ### Known issues
 
-- OpenVINO(==2022.3) IR inference is not working well on 2-stage models (e.g. Mask-RCNN) exported from torch==1.13.1
+- OpenVINO(==2023.0) IR inference is not working well on 2-stage models (e.g. Mask-RCNN) exported from torch==1.13.1
```
Comment (goodsong81) [inline CHANGELOG.md:34]: Oh, is it still not working? @jaegukhyun What would you say?
The Platform team has been waiting for this update for torch1.13 update.

**Review by jaegukhyun — APPROVED** [review]

File: src/otx/algorithms/detection/adapters/openvino/task.py [hunk 19]
```diff
@@ -468,8 +449,15 @@ def load_inferencer(
...
         if self.task_type == TaskType.DETECTION:
+            if (
+                self.task_environment.model_template.model_template_id == "Custom_Object_Detection_YOLOX"
+                and not self.config.tiling_parameters.enable_tiling
+            ):
+                args.append({"resize_type": "fit_to_window_letterbox", "pad_value": 114})
             inferencer: BaseInferencerWithConverter = OpenVINODetectionInferencer(*args)
         if self.task_type == TaskType.INSTANCE_SEGMENTATION:
+            if self.config.tiling_parameters.enable_tiling:
+                args.append({"resize_type": "fit_to_window_letterbox"})
```
Comment (goodsong81) [inline src/otx/algorithms/detection/adapters/openvino/task.py:460]: These setting might be decided from preprocessing options in the future :)

**Review by goodsong81 — APPROVED** [review]
Thank you, it's a huge update!

File: CHANGELOG.md (hunk 18 above)
Comment (jaegukhyun) [reply CHANGELOG.md:34]: For large dataset it works well, but in some medium and small datasets especially toy datasets, it shows degraded performance.

Comment (jaegukhyun) [reply CHANGELOG.md:34]: At now, we still skip e2e tests for action detection and instance segmentation

Comment (goodsong81) [reply CHANGELOG.md:34]: It means that it's not an torch1.13 vs OV incompatibility anymore, but still has accuracy issue. Thank you for clasification.

File: src/otx/algorithms/detection/adapters/openvino/task.py (hunk 19 above)
Comment (sovrasov) [reply src/otx/algorithms/detection/adapters/openvino/task.py:460]: Right now we have no an appropriate mechanism to forward such parameters from algo backend config to inference, but when we fully switch to ModelAPI, that will be moved to the task.export() level. However, that's not backward compatible. Or which mechanism do you mean?
